python/python/Ice/Object.py

```diff
@@ -84,76 +81,6 @@ def ice_staticId():
         """
         return "::Ice::Object"
 
-    def _iceDispatch(self, cb, method, args):
-        # Invoke the given servant method. Exceptions can propagate to the caller.
-        result = method(*args)
-
-        # If the result is a coroutine and the communicator has a custom coroutine executor, execute the coroutine using
-        # the configured executor. The returned future is then used to wait for dispatch completion.
-        if inspect.iscoroutine(result):
-            assert (
-                len(args) > 0
-            ), "args must have at least one element representing the dispatch current parameter"
-            current = args[-1]
-            coroutineExecutor = current.adapter.getCommunicator().getCoroutineExecutor()
-            if coroutineExecutor:
-                result = coroutineExecutor(result)
-                if not isinstance(result, Future) and not callable(getattr(result, "add_done_callback", None)):
-                    raise TypeError("Executor must return a Future-like object")
-
-        # Check for a future.
-        if isinstance(result, Future) or callable(
-            getattr(result, "add_done_callback", None)
-        ):
-
-            def handler(future):
-                try:
-                    cb.response(future.result())
-                except Exception:
-                    cb.exception(sys.exc_info()[1])
-
-            result.add_done_callback(handler)
-        elif inspect.iscoroutine(result):
-            self._iceDispatchCoroutine(cb, result)
-        else:
-            cb.response(result)
-
-    def _iceDispatchCoroutine(self, cb, coro, value=None, exception=None):
-        try:
-            if exception:
-                result = coro.throw(exception)
-            else:
-                result = coro.send(value)
-
-            if result is None:
-                # The result can be None if the coroutine performs a bare yield (such as asyncio.sleep(0))
-                cb.response(None)
```

Comment:
This was bogus `await asyncio.sleep(0)` would indeed return `None` but doesn't mean dispatch completed, and we should not be calling `cb.respose` in this case.

In real life this code should not be used when handling asyncio coroutines, you should have a coroutineExecutor in place.

I tested we were never exercising this branch with the test suite. Is gone from the new implementation.